src/backend/distributed/utils/colocation_utils.c

```diff
@@ -431,11 +431,11 @@ ShardsIntervalsEqual(ShardInterval *leftShardInterval, ShardInterval *rightShard
 		return false;
 	}
 
-	if (leftIntervalPartitionMethod == DISTRIBUTE_BY_HASH)
+	if (IsCitusTableType(leftShardInterval->relationId, HASH_DISTRIBUTED))
 	{
 		return HashPartitionedShardIntervalsEqual(leftShardInterval, rightShardInterval);
 	}
-	else if (leftIntervalPartitionMethod == DISTRIBUTE_BY_NONE)
+	else if (IsCitusTableType(leftShardInterval->relationId, REFERENCE_TABLE))
```

Comment:
I guess we're good to use `NO_DIST_KEY` here, because they also have single shard